File: Smallest number with sum of digits as N and divisible by 10^N.py
class Solution:
	def digitsNum(self, N):
		# Code here
		ans = ""
		if N == 0:
		    ans += "0"
		    return ans
		
		if (N%9 != 0):
		    ans += chr(N%9 + ord("0"))
		
		for i in range(1, (N//9) + 1):
		    ans += "9"
		
		for i in range(1, N+1):
		    ans += "0"
		
		return ans

File: Smallest number with sum of digits as N and divisible by 10^N.cpp
class Solution{
	public:
	string digitsNum(int N)
	{
	    // Code here.
	    string ans = "";
	    int n = N;
	    
	    while(n >= 9)
	    {
	        n -= 9;
	        ans.push_back('9');
	    }
	    
	    if(n)
	        ans.push_back('0'+n);
	    
	    reverse(ans.begin(), ans.end());
	    
	    for(int i=0; i<N; i++)
	    ans.push_back('0');
	    
	    return ans;
	}
};
